packages/notebook/src/widget.ts

```diff
@@ -418,6 +424,17 @@ export class StaticNotebook extends WindowedList<NotebookViewModel> {
         // @ts-expect-error Cell has no index signature
         newCell[state] = view[state];
       }
+
+      if (from > to) {
+        if (this.widgets[to + i].model.type === 'code') {
+          (this.widgets[to + i].model as CodeCellModel).isDirty = dirtyState[i];
+        }
+      } else {
+        if (this.widgets[to + i - n + 1].model.type === 'code') {
+          (this.widgets[to + i - n + 1].model as CodeCellModel).isDirty =
+            dirtyState[i];
+        }
```

Comment:
I wonder why is it that `rendered` and `headingCollapsed` do not need `- n + 1` but `isDirty` needs it? It seems a bit suspicious to me that the implementation here differs from the one for markdown cell properties. I suspect that one of those is wrong.